sdks/java/core/src/main/java/org/apache/beam/sdk/io/Source.java

```diff
@@ -64,6 +66,7 @@
   /**
    * Returns the default {@code Coder} to use for the data read from this source.
    */
+  @Nullable
```

Comment:
You've got a point about the consistency. In that case I see it like this:

Source.java
```
public Coder<T> getDefaultOutputCoder() throws CannotProvideCoderException {
    throw new CannotProvideCoderException("Source.getDefaultOutputCoder called.");
  }
```

As the exception is checked, I have the following questions:

1. Should I throw further (throws clause) the exception in every source that extends the Source class  (CompressedSource, MicrobatchSource, etc)? Because it seems that there is no way to infer/know what Coder should it be. If not, could you give me an example of how should I handle it?

2. Is it ok that it's a breaking change to the API? There are some classes that are not marked as experimental. eg. Read.Bounded, like below: 

Read.java
```
    @Override
    protected Coder<T> getDefaultOutputCoder() throws CannotProvideCoderException {
      return source.getDefaultOutputCoder(); 
    }

    @Override
    public final PCollection<T> expand(PBegin input) throws CannotProvideCoderException {
      source.validate();

      return PCollection.<T>createPrimitiveOutputInternal(input.getPipeline(),
          WindowingStrategy.globalDefault(), IsBounded.BOUNDED)
          .setCoder(getDefaultOutputCoder());
    }
```